src/common/aes/aes.h

```diff
@@ -95,19 +107,30 @@ void OQS_AES256_ECB_enc(const uint8_t *plaintext, const size_t plaintext_len, co
 void OQS_AES256_ECB_enc_sch(const uint8_t *plaintext, const size_t plaintext_len, const void *schedule, uint8_t *ciphertext);
 
 /**
- * AES counter mode keystream generator.  A scheduled generated by
+ * AES counter mode keystream generator.  A context generated by
  * OQS_AES256_CTR_load_schedule() is passed rather then a key.
  *
  * Handles a 12- or 16-byte IV.  If a 12-byte IV is given, then 4 counter
  * bytes are initialized to all zeros.
  *
  * @param iv       12- or 16-byte initialization vector.
  * @param iv_len   Lengh of IV in bytes.
- * @param schedule Abstract data structure for a key schedule.
+ * @param ctx      Abstract data structure for a key schedule.
+ * @param out      Pointer to a block of memory which is big enough to contain out_len bytes; the result will be written here.
+ * @param out_len  Length of output bytes to generate.
+ */
+void OQS_AES256_CTR_sch(const uint8_t *iv, size_t iv_len, const void *ctx, uint8_t *out, size_t out_len);
+
+/**
+ * AES counter mode keystream generator. A context generated by
+ * OQS_AES256_CTR_load_schedule() and OQS_AES256_CTR_load_nonce() is passed
+ * rather than a key and an IV.
+ *
+ * @param ctx      Abstract data structure for key schedule and IV.
  * @param out      Pointer to a block of memory which is big enough to contain out_len bytes; the result will be written here.
  * @param out_len  Length of output bytes to generate.
  */
-void OQS_AES256_CTR_sch(const uint8_t *iv, size_t iv_len, const void *schedule, uint8_t *out, size_t out_len);
+void OQS_AES256_CTR_sch_ivinit(void *ctx, uint8_t *out, size_t out_len);
```

Comment:
Why is this called `OQS_AES256_CTR_sch_ivinit`? The `ivinit` part of it makes it sound like it's going to initialize the IV, but in fact it generates output...

Reply:
Right, it's actually an update function since it internally updates the IV/counter and allows to be called multiple times. I've now renamed it to `OQS_AES256_CTR_sch_upd_blks`.

Reply:
The API naming is now more in line with the "incremental" API from SHA2/3. (see PR comments)